applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -213,28 +212,50 @@ Vector& ElasticIsotropic3D::CalculateValue(
     if (rThisVariable == STRAIN ||
         rThisVariable == GREEN_LAGRANGE_STRAIN_VECTOR ||
         rThisVariable == ALMANSI_STRAIN_VECTOR) {
-        this->CalculateCauchyGreenStrain( rParameterValues, rValue);
+
+        Flags& r_flags = rParameterValues.GetOptions();
+
+        // Previous flags saved
+        const bool flag_const_tensor = r_flags.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR );
+        const bool flag_stress = r_flags.Is( ConstitutiveLaw::COMPUTE_STRESS );
+
+        // Set flags to only compute the strain
+        r_flags.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
+        r_flags.Set(ConstitutiveLaw::COMPUTE_STRESS, false);
+
+        ElasticIsotropic3D::CalculateMaterialResponsePK2(rParameterValues);
+        rValue = rParameterValues.GetStrainVector();
+
+        // Previous flags restored
+        r_flags.Set( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, flag_const_tensor );
+        r_flags.Set( ConstitutiveLaw::COMPUTE_STRESS, flag_stress );
```

Comment:
can you please explain why this is changed?

Reply:
Idem

Reply:
I think that now, with the inclusion fo the initial strain, since the addition of this initial strain is done inside the calculateMaterialResponse, in order to print properly the strain is required to call the CalcMatResp....

Reply:
The original code directly jumping to the calculation of the strain in the CL. The code flow should be to check the USE_ELEMENT_PROVIDED_STRAIN flag, compute the strain in the CL if false, and increment it with the initial strain (if provided). This is already coded in the CalulateMaterialResponsePK2 function, so we are getting the strain from there, similarly to the way the stress is computed. Also not checking the USE_ELEMENT_PROVIDED_STRAIN flag was unexpected in the SmallStrainIsotropicDamagage3D CL and was throwing segfaults.

Reply:
(btw, GREEN_LAGRANGE_STRAIN_VECTOR and ALMANSI_STRAIN_VECTOR never reach this point, as they are intercepted in the base solid element. A fix to that requires fixing also some tests, as @loumalouomega  noted in #8183)